robottelo/ui/bookmark.py

```diff
@@ -0,0 +1,94 @@
+# -*- encoding: utf-8 -*-
+"""Implements Bookmarks UI"""
+from robottelo.decorators import bz_bug_is_open
+from robottelo.ui.base import Base, UIError
+from robottelo.ui.locators import common_locators, locators
+from robottelo.ui.navigator import Navigator
+
+
+class Bookmark(Base):
+    """Implements CRUD functions for UI"""
+
+    def navigate_to_entity(self):
+        """Navigate to Bookmark entity page"""
+        Navigator(self.browser).go_to_bookmarks()
+
+    def _search_locator(self):
+        """Specify locator for Bookmark entity search procedure"""
+        return locators['bookmark.select_name']
+
+    def update(self, controller, name, new_name=None, new_query=None,
+               new_public=None):
+        """Updates a bookmark."""
+        element = self.search(controller, name)
+        self.click(element)
+        if new_name:
+            self.wait_until_element(locators['bookmark.name'])
```

Comment:
Sure, anyway waiter is called for all types of elements, so you just calling them two times.
And btw you did not do that in your base class method for the same functionality :)